api/src/main/java/com/graphhopper/internal/LazyPointList.java

```diff
@@ -0,0 +1,152 @@
+package com.graphhopper.internal;
+
+
+import com.graphhopper.bean.RoutePoint;
+
+import java.util.*;
+
+public class LazyPointList implements List<RoutePoint> {
```

Comment:
Can we avoid Lazy naming here? I thought about lazy loading, but you meant lazy iteration pattern?